starter-code/scripts/models/article.js

```diff
@@ -1,101 +1,126 @@
+        success: function(data, message, xhr) {
+          var eTag = xhr.getResponseHeader('eTag');
+          if (!localStorage.eTag || eTag !== localStorage.eTag) {
+            Article.getAll(nextFunction); // DONE: pass 'nextFunction' into Article.getAll();
+          } else {
+            Article.loadAll(JSON.parse(localStorage.hackerIpsum));
+            // DONE: Replace the following line with 'nextFunction' and invoke it!
+            nextFunction();
+          }
+        }
+      });
+    } else {
+      Article.getAll(nextFunction); // DONE: pass 'nextFunction' into getAll();
+    }
+  };
 
-Article.numWordsAll = function() {
-  return Article.allArticles.map(function(currentArticle) {
-    return currentArticle.body.match(/\w+/g).length;
-  }).reduce(function() {
-    // TODO: complete this function to sum up all of the words.
-  })
-}
+  Article.getAll = function(nextFunction) {
+    $.getJSON('/data/hackerIpsum.json', function(responseData, message, xhr) {
+      localStorage.eTag = xhr.getResponseHeader('eTag');
+      Article.loadAll(responseData);
+      localStorage.hackerIpsum = JSON.stringify(responseData);
+      // DONE invoke our parameter.
+      nextFunction();
+    });
+  };
 
-Article.allAuthors = function() {
-  // TODO: return a mapped collection
-      // with just the author names
+  Article.numWordsAll = function() {
+    return Article.allArticles.map(function(currentArticle) {
+      return currentArticle.body.match(/\w+/g).length;
+    })
+    .reduce(function(acc, curr) {
+      //console.log(acc);
+      // TODO: complete this function to sum up all of the words.
+      return acc + curr;
+    }, 0);
+  }
 
+  Article.allAuthors = function() {
+    // TODO: return a mapped collection
+    // with just the author names ...workout reduce
+    //console.log(Article.allArticles)
+    return Article.allArticles.map(function(article) {
+      return article.author;
+    })
+    .reduce(function(accum, current) {
+      if (accum.indexOf(current) === -1) {
+        accum.push(current)
+      }
       //then chain reduce, and set the accumulator to an array
       // to build a unique list of author names.
-}
+      return accum;
+    }, [])
+  }
 
-Article.numWordsByAuthor = function() {
-  // TODO: transform each author element into an object with 2 properties:
+  Article.numWordsByAuthor = function() {
+    // TODO: transform each author element into an object with 2 properties:
     // one for the author's name, and one for the total number of words
     // written by the specified author.
     return Article.allAuthors().map(function(currentAuthor) {
+      console.log(currentAuthor)
       return {
         name: currentAuthor,
-        numWords: // someCollection.filter(function(curArticle) {
-          // what do we return here to check for matching authors?
+        numWords:
+        Article.allArticles.filter(function(curArticle) {
+          return curArticle.author === currentAuthor;
```

Comment:
Great work limiting the articles of interest to those written by the current Author!